src/solver/optim-model-filler/PartialKey.cpp

```diff
@@ -25,31 +25,33 @@
 
 namespace Antares::Optimization
 {
-
-// PartialKey
-PartialKey::PartialKey(const std::string& component_id, const std::string& variable_id):
-    component_id(component_id),
-    variable_id(variable_id)
+PartialKey::PartialKey(std::string_view component_id, std::string_view variable_id) noexcept:
+    component_id_(interner().intern(component_id)),
+    variable_id_(interner().intern(variable_id))
 {
 }
 
-const std::string& PartialKey::getComponent() const
+const std::string& PartialKey::getComponent() const noexcept
 {
-    return component_id;
+    return interner().get(component_id_);
 }
 
-const std::string& PartialKey::getVariable() const
+const std::string& PartialKey::getVariable() const noexcept
 {
-    return variable_id;
+    return interner().get(variable_id_);
 }
 
-// PartialKeyhash
+PartialKey::StringInterner& PartialKey::interner() noexcept
+{
+    static StringInterner inst;
+    return inst;
```

Comment:
Global variable here, it may be better to have one instance for each `VariableDictionary`